staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -118,6 +119,14 @@ func (c *tlsTransportCache) get(config *Config) (http.RoundTripper, error) {
 		DisableCompression:  config.DisableCompression,
 	})
 
+	if config.Proxy != nil {
+		// DialTLSContext provides an optional dial function for non-proxied HTTPS requests.
+		// The function will be used for connections to the proxy if the proxy itself is using HTTPS.
+		// In this case TLSClientConfig and TLSHandshakeTimeout will be ignored for connections to the proxy.
+		// We do this because we don't want options such as tls-server-name to be applied to the proxy connection.
+		transport.DialTLSContext = (&tls.Dialer{}).DialContext
```

Comment:
setting both this and DialContext is really confusing (and doesn't play nicely with helpers like `DialerFor(transport http.RoundTripper) (DialFunc, error)` that only pull out the DialContext method, since that's the one we set)

Reply:
setting this also bypasses/ignores some of the TLS and connection-relevant options we're setting ... timeout/keepalive dialer options, TLSHandshakeTimeout, and maybe CA bundle and insecure settings